test/logging/log.js

```diff
@@ -0,0 +1,534 @@
+
+      Log.assignSeverityToEntries_(originalEntries, SEVERITY);
+
+      assert.deepEqual(originalEntries, ENTRIES);
+    });
+  });
+
+  describe('createWriteStream', function() {
+    beforeEach(function() {
+      log.request = util.noop;
+    });
+
+    it('should return a writable object stream', function() {
+      var ws = log.createWriteStream();
+
+      assert.strictEqual(ws.writable, true);
+    });
+
+    it('should make a request once writing started', function(done) {
+      log.request = function(reqOpts) {
+        assert.strictEqual(reqOpts.method, 'POST');
+        assert.strictEqual(reqOpts.uri, '/entries:write');
+
+        setImmediate(done);
+
+        return through();
+      };
+
+      var ws = log.createWriteStream();
+      ws.emit('writing');
+    });
+
+    it('should emit the response from the request', function(done) {
+      var response = {};
+      var ws = log.createWriteStream();
+
+      log.request = function() {
+        var stream = through();
+
+        setImmediate(function() {
+          stream.emit('response', response);
+        });
+
+        return stream;
+      };
+
+      ws.on('response', function(response_) {
+        assert.strictEqual(response_, response);
+        done();
+      });
+
+      ws.emit('writing');
+    });
+
+    it('should format each entry', function(done) {
+      var entry = { formatted: false };
+      var formattedEntry = { formatted: true };
+
+      var ws = log.createWriteStream();
+
+      var expectedData = {
+        entries: [
+          formattedEntry,
+          formattedEntry
+        ]
+      };
+
+      var requestStream = concat(function(data) {
+        assert.deepEqual(JSON.parse(data), expectedData);
+        done();
+      });
+
+      log.request = function() {
+        return requestStream;
+      };
+
+      log.formatEntry_ = function(entry_) {
+        assert.strictEqual(entry_, entry);
+        return formattedEntry;
+      };
+
+      ws.write(entry);
+      ws.end(entry);
+    });
+  });
+
+  describe('insert', function() {
+    var ENTRY = {};
+    var LABELS = {};
+
+    it('should not require labels', function(done) {
+      log.request = function(reqOpts) {
+        assert.deepEqual(reqOpts.json.commonLabels, {});
+        done();
+      };
+
+      log.insert(ENTRY, assert.ifError);
+    });
+
+    it('should make the correct API request', function(done) {
+      var formattedEntry = {};
+
+      log.formatEntry_ = function() {
+        return formattedEntry;
+      };
+
+      log.request = function(reqOpts) {
+        assert.strictEqual(reqOpts.method, 'POST');
+        assert.strictEqual(reqOpts.uri, '/entries:write');
+        assert.strictEqual(reqOpts.json.commonLabels, LABELS);
+        assert.strictEqual(reqOpts.json.entries[0], formattedEntry);
+
+        done();
+      };
+
+      log.insert(ENTRY, LABELS, assert.ifError);
+    });
+
+    it('should exec callback with only error and API response', function(done) {
```

Comment:
Sure, specifically, this line: https://github.com/stephenplusplus/gcloud-node/blob/310017043fba58daad17834e0b95cff9ba615229/lib/logging/log.js#L347 -- we want to make sure *only* two total arguments are passed to the user and they are the error and response body from the API request.